core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -377,20 +384,23 @@ protected long getBlockSize(long pos) {
    * Checks whether block instream and cache outstream should be updated. This function is only
    * called by {@link #updateStreams()}.
    *
-   * @param currentBlockId cached result of {@link #getCurrentBlockId()}
    * @return true if the block stream should be updated
    */
-  protected boolean shouldUpdateStreams(long currentBlockId) {
-    if (mCurrentBlockInStream == null || currentBlockId != mStreamBlockId) {
-      return true;
+  protected boolean shouldUpdateStreams() {
+    if (mBlockIdIndex >= mBlockIds.size()) {
+      // EOF
+      return false;
     }
+    return mCurrentBlockInStream == null || mCurrentBlockInStream.remaining() == 0;
+  }
+
+  private void assureCacheStreamInSync() {
```

Comment:
nit: name this after `checkCacheStreamInSync` to be consistent (we say check xxx in this code base)

Reply:
done